core/include/drivers/rng_pta_client.h

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2021, Foundries.io Ltd
+ */
+
+#ifndef __RNG_PTA_CLIENT_H
+#define __RNG_PTA_CLIENT_H
+
+#define PTA_RNG_UUID { 0x035a4479, 0xc369, 0x47f4, \
+		{ 0x94, 0x51, 0xc6, 0xfd, 0xff, 0x28, 0xad, 0x65 } }
```

Comment:
linux since v5.1 supports PTA RNG with another UUID [1]. This change should move synquacer PTA [2] to this generic source path.

[1] https://elixir.bootlin.com/linux/v5.11/source/drivers/char/hw_random/optee-rng.c#L274
[2] https://elixir.bootlin.com/op-tee/3.12.0/source/core/arch/arm/plat-synquacer/rng_pta_client.h#L9

Reply:
@etienne-lms I think having different UUIDs for different pseudo TAs makes sense. We just need to append optee-rng device UUID table [1] with this another UUID.

And for the interface (rng_pta_client.h), I agree with you that we should move that to common location.

[1] https://elixir.bootlin.com/linux/v5.11/source/drivers/char/hw_random/optee-rng.c#L274

Reply:
These PTAs do the same thing, no? If so, why 2 UUID?

despite the fact that I agree the way you plugged enumerated devices makes it so easy adding an OP-TEE service UUID in linux :)

Reply:
> These PTAs do the same thing, no? If so, why 2 UUID?

I would say these two pTAs communicates with optee-rng driver using common interface but underlying backend to fetch entropy is different which may differ with respect to speed, quality etc. IMO, its better to have a unique device ID (UUID) to differentiate among the two. It is somewhat similar to having unique device identities on say I2C bus etc.